src/Microsoft.Extensions.Testing.Abstractions/Messages/Message.cs

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Extensions.Testing.Abstractions
+{
+    public class Message
+    {
+        public string MessageType { get; set; }
+
+        public JToken Payload { get; set; }
```

Comment:
Generally it reduces parsing overhead. The JSON parser has already parsed the entire message to deserialize it into the `Message` class. Storing it as a string means that when `ToObject<T>` is called (or equivalent), the string has to be re-parsed and then mapped to the `T` object. When it's a `JToken`, the parsing is still done and the data just has to be mapped.

It's a micro-optimization for sure and we could probably refine this code, but I'd agree with @piotrpMSFT that leaving the DNX code as-is for now is a good idea to ensure we aren't missing something.